### PR TITLE
[Refactor] Refine rpc cache configs

### PR DESCRIPTION
**Pull request by FMX**
# [Reactor]Refine rpc cache configs.

### How was this patch tested?

/cc @related-reviewer

/assign @main-reviewer

### CODE REVIEW

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 1]
```diff
@@ -1985,16 +1985,16 @@ object CelebornConf extends Logging {
       .createWithDefault(256)
 
   val RPC_CACHE_CONCURRENT_LEVEL: ConfigEntry[Int] =
-    buildConf("celeborn.rpcCache.concurrent.level")
+    buildConf("celeborn.rpc.cache.concurrent.level")
```
Comment (pan3793) [inline common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:1988]: concurrentLevel

Comment (FMX) [reply common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:1988]: Check out the spark config, I think there is no need to be concurrentLevel.

>   private[spark] val IO_COMPRESSION_ZSTD_LEVEL =
>     ConfigBuilder("spark.io.compression.zstd.level")
>       .doc("Compression level for Zstd compression codec. Increasing the compression " +
>         "level will result in better compression at the expense of more CPU and memory")
>       .version("2.3.0")
>       .intConf
>       .createWithDefault(1)

Comment (pan3793) [reply common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:1988]: IMO, it's not the same case. Zstd is one of the supported compression types, and `level` is a property of zstd. "concurrent level" should be a property for "cache"

Comment (FMX) [reply common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:1988]: Reasonable.

**Review by pan3793 — APPROVED** [review]

**Pull request merged**